src/main/java/org/jabref/logic/cleanup/DoiCleanup.java

```diff
@@ -57,14 +59,25 @@ public List<FieldChange> cleanup(BibEntry entry) {
         } else {
             // As the Doi field is empty we now check if note, url, or ee field contains a Doi
             for (Field field : FIELDS) {
-                Optional<DOI> doi = entry.getField(field).flatMap(DOI::parse);
+                Optional<String> fieldContentOpt = entry.getField(field);
+
+                Optional<DOI> doi = fieldContentOpt.flatMap(DOI::parse);
 
                 if (doi.isPresent()) {
                     // Update Doi
                     Optional<FieldChange> change = entry.setField(StandardField.DOI, doi.get().getDOI());
                     change.ifPresent(changes::add);
                     removeFieldValue(entry, field, changes);
                 }
+
+                if (StandardField.EPRINT == field && fieldContentOpt.isPresent()) {
```

Comment:
I think the isPresent check for fieldContentOPt is not necessary here as it's already covered by the ifPresent

Reply:
Nice catch, it's removed now.